container.go

```diff
@@ -305,30 +307,90 @@ func (c *ContainerRequest) GetTag() string {
 	return strings.ToLower(t)
 }
 
-// Deprecated: Testcontainers will detect registry credentials automatically, and it will be removed in the next major release
-// GetAuthConfigs returns the auth configs to be able to pull from an authenticated docker registry
+// Deprecated: Testcontainers will detect registry credentials automatically, and it will be removed in the next major release.
+// GetAuthConfigs returns the auth configs to be able to pull from an authenticated docker registry.
+// Panics if an error occurs.
 func (c *ContainerRequest) GetAuthConfigs() map[string]registry.AuthConfig {
-	return getAuthConfigsFromDockerfile(c)
+	auth, err := getAuthConfigsFromDockerfile(c)
+	if err != nil {
+		panic(fmt.Sprintf("failed to get auth configs from Dockerfile: %v", err))
```

Comment:
Not sure I understand, could you clarify?

Reply:
Yeah, this method will be removed in the next major version, so in this change I'd move the panic to the `getAuthConfigsFromDockerfile` method. Or maybe I'm wrong and we want to panic here because it's deprecated to inform that something will eventually change 🤔 Is that a pattern for deprecations that I'm not aware of? Just curious to learn about that.

Reply:
It only needs to panic here because the internal method now returns an error which this method can't handle without an API breaking change.

We could ignore like before, but thought it better to have the user of deprecated method know there was an issue.

Reply:
I see, thanks for the clarification. Given the comment explicitly mentions it, lgtm